java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1187,6 +1293,52 @@ public final Builder append(double value) {
       return this;
     }
 
+    public final Builder append(BigDecimal value) {
+      return append(value, null);
+    }
+
+    //
+    public final Builder append(BigDecimal value, RoundingMode roundingMode) {
+      assert type.isDecimalType();
+      assert currentIndex < rows;
+      BigInteger unscaledValue;
+      // rescaled input decimal if RoundingMode set
+      if (roundingMode == null) {
+        assert -type.getScale() == value.scale();
```

Comment:
Why not treat `roundingMode == null` the same as `roundingMode == BigDecimal.ROUNDING_UNNECESSARY` here?

Reply:
For `ROUNDING_UNNECESSARY`,  rescaling can still happen if target scale is larger than current scale (by definition of `BigDecimal`). But at here, we don't want to perform rescaling under any conditions. Because the most important usage of method `append(scale, null)` is  `fromDecimals(BigDecimal... values)`, in which we assume (require) all input BigDecimals sharing same scale.

Reply:
Why do we require all decimals need to be the same scale at this point?  If we can safely scale the value without losing any precision (i.e.: the behavior when specifying ROUNDING_UNNECESSARY), why not do so automatically?  `BigDecimal` will throw if it can't adjust the scale without losing precision, so it's essentially the same type of assert but one that won't throw in an unnecessary situation.